### ISSUE TITLE
Bring the SPO to operatorhub.io

### ISSUE THREAD

**Issue by saschagrunert** [kubernetes-sigs/security-profiles-operator#779]
#### What would you like to be added:
Adding the operator to [Operator Hub](https://operatorhub.io/) to automate the lifecycle of the operator when installed in OpenShift.

#### Why is this needed:

It would be a good addition to spread the usage of the operator while making its installation and upgrade easier.

#### User story covered

None so far.

**Comment by jhrozek** [comment]
Hmm, OK, I started playing with the refactoring and I think we wouldn't have to touch the code itself much if at all. What the tools that generate bundles really care about are the APIs and the manifests. So if we moved the api structure to the one that matches kubebuilder's and moved manifests from deploy to config, that /should/ be it. I'll try to make some simple protorype, even if it didn't pass all the tests so that we can discuss easier.
